### PR TITLE
Split start.sh to start.sh and pre-start.sh

### PR DESCRIPTION
**Pull request by AjaCoder**
Split start.sh to start.sh and pre-start.sh for easy support of base image usage with a different start action.
With this change, anyone can create an image using from and override the start.sh script with their own script. The only thing needed is to call pre-start.sh script from the overriding script.

### CODE REVIEW

File: scripts/pre-start.sh [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+	
+	mkdir -p /usr/local/var/lib/gvm/data-objects
+	
+	ln -s /data/data-objects /usr/local/var/lib/gvm/data-objects
+	
+	chown gvm:gvm -R /data/data-objects
+	chown gvm:gvm -R /usr/local/var/lib/gvm/data-objects
+fi
+
+if [ AUTO_SYNC ] || [ ! -f "/firstsync" ]; then
```
Comment (FloThinksPi) [inline scripts/pre-start.sh:242]: I especially like the env flag to skip sync. Usecase can be sending an appliance to a customer for one time analysis. As internet access is likely not given, the container currently wont start and abort at the failed sync. With this one can start the container with a flag to skip that step 👍 

However as the PR suggests this is not a simple split into 2 files(there are additions).
Needs also a rebase and gitbook doc. Would be cool if we can get this merged.

Comment (austinsonger) [reply scripts/pre-start.sh:242]: Sorry, someone else hasn't reviewed this yet. I will take a look.

Comment (austinsonger) [reply scripts/pre-start.sh:242]: I'm gonna merge this on the `dev` branch and work on that branch after merge.

**Review by austinsonger — APPROVED** [review]

**Comment by FloThinksPi** [comment]
Oh i forgot to comment here further. I tried out this change and it didnt work for me. Strangely i got syntax errors where the new Env variable got compared. So i reworked it and i am using my fork currently.

**Comment by AjaCoder** [comment]
The issue is probably that since I have published the PR the code base had changed.

**Comment by AjaCoder** [comment]
Closing the PR, I will open again when fixed.

**Pull request closed without merging**